cpu/nrf53/include/periph_cpu.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2023 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup         cpu_nrf53
+ * @{
+ *
+ * @file
+ * @brief           nRF5340 specific definitions for handling peripherals
+ *
+ * @author          Dylan Laduranty <dylan.laduranty@mesotic.com>
+ */
+
+#ifndef PERIPH_CPU_H
+#define PERIPH_CPU_H
+
+#include "periph_cpu_common.h"
+#include "macros/units.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   System core clock speed, for all NRF53 Application core.
+ */
+#define CLOCK_CORECLOCK     MHZ(128)
+
+/**
+ * @brief   Structure for UART configuration data
+ */
+typedef struct {
+    NRF_UARTE_Type *dev;    /**< UART with EasyDMA device base
+                             * register address */
+    gpio_t rx_pin;          /**< RX pin */
+    gpio_t tx_pin;          /**< TX pin */
+#ifdef MODULE_PERIPH_UART_HW_FC
+    gpio_t rts_pin;         /**< RTS pin */
+    gpio_t cts_pin;         /**< CTS pin */
+#endif
+    uint8_t irqn;           /**< IRQ channel */
+} uart_conf_t;
```

Comment:
Any reason why this is not in `periph_cpu_common.h`?

Reply:
I just follow what's existing in our codebase. For now each nRF families define its own struct.
I can check if  I can move all structs to `periph_cpu_common.h` but I would prefer in another PR. I already need to rework the UART IRQ management for nRF53 and nRF9160 because of changes introduce by Nordic for these families. (and to add I2C/SPI at the same time) so I can add this specific point to this PR instead :)

Reply:
Except nRF51 and NRF52832XXAA ;)
I'll move them in a separate commit during the next round of fixup commits.

Reply:
Moved.